core-utilities/src/main/java/edu/illinois/cs/cogcomp/annotation/XmlTextAnnotationMaker.java

```diff
@@ -0,0 +1,74 @@
+/**
+ * This software is released under the University of Illinois/Research and Academic Use License. See
+ * the LICENSE file in the root folder for details. Copyright (c) 2016
+ *
+ * Developed by: The Cognitive Computation Group University of Illinois at Urbana-Champaign
+ * http://cogcomp.cs.illinois.edu/
+ */
+package edu.illinois.cs.cogcomp.annotation;
+
+import edu.illinois.cs.cogcomp.core.datastructures.IntPair;
+import edu.illinois.cs.cogcomp.core.datastructures.Pair;
+import edu.illinois.cs.cogcomp.core.datastructures.textannotation.TextAnnotation;
+import edu.illinois.cs.cogcomp.core.datastructures.textannotation.XmlTextAnnotation;
+import edu.illinois.cs.cogcomp.core.utilities.StringTransformation;
+import edu.illinois.cs.cogcomp.core.utilities.TextCleanerStringTransformation;
+import edu.illinois.cs.cogcomp.core.utilities.XmlDocumentProcessor;
+
+import java.util.Map;
+import java.util.Set;
+
+/**
+ * Instantiates a XmlTextAnnotation object from xml text. The xml is parsed into body text (which is further cleaned
+ *    up as needed), and this cleaned text is used to create a TextAnnotation. Additional information is extracted
+ *    from the xml source. The mapping between the xml source and the cleaned text (i.e. mapping between character
+ *    offsets) is also derived.
+ * The goal is to provide text that can be processed easily with an NLP pipeline without a lot of hacks to work around
+ *    ill-formatted text. The annotations so produced can then be mapped to the offsets in the original xml text,
+ *    and combined with supplementary information extracted from the xml markup.
+ *
+ * @author mssammon
+ */
+
+public class XmlTextAnnotationMaker {
+
+    private final TextAnnotationBuilder taBuilder;
+    private final XmlDocumentProcessor xmlProcessor;
+
```

Comment:
What are the rules for the use of blank lines? Where do we use one, where do we use two and where do we use three? And why?